osu.Game/Screens/OnlinePlay/DailyChallenge/DailyChallengeIntro.cs

```diff
@@ -461,6 +464,8 @@ private void beginAnimation()
                         {
                             Schedule(() =>
                             {
+                                statics.SetValue(Static.DailyChallengeIntroPlayed, true);
```

Comment:
Not sure about this flag being set at the very end of the animation rather than at the start already.

Reply:
I imagined some users might find it more convenient to press the daily challenge screen to intro the screen then immediately hit back then go again, and I don't know if I want to allow that.

Reply:
cc @peppy

Reply:
Should be set after playing out.

> I imagined some users might find it more convenient to press the daily challenge screen to intro the screen then immediately hit back then go again, and I don't know if I want to allow that.

Sounds like really bad UX. If users hate the intro that much then we've done something wrong. We don't create hacky UX workaround like this intentionally.

Reply:
Yeah that's basically what I'm trying to say, that we don't want that to happen.